src/realm/sync/changeset_parser.cpp

```diff
@@ -30,6 +30,9 @@ struct ChangesetParser::State {
 
     StringBuffer m_buffer;
     util::metered::set<uint32_t> m_valid_interned_strings;
```

Comment:
I am not sure how having the vector sorted by the intern index will help. `std::find_if` and `emplace` have linear complexity (emplace could be constant if the intern strings are already sorted by their index). I agree it would use less memory though. Given the low number of intern strings, that should not be a problem.

I would have normally used an `unordered_set` but it does not have support for heterogenous lookups until C++20.